code/modules/vehicles/mecha/equipment/mecha_equipment.dm

```diff
@@ -102,52 +87,81 @@
 	var/C = chassis.loc
 	chassis.use_power(energy_drain)
 	. = do_after(user, equip_cooldown, target=target, interaction_key = interaction_key)
-	if(!chassis || chassis.loc != C || src != chassis.selected || !(get_dir(chassis, target)&chassis.dir))
+	if(!chassis || chassis.loc != C || !(get_dir(chassis, target)&chassis.dir))
 		return FALSE
 
 /obj/item/mecha_parts/mecha_equipment/proc/do_after_mecha(atom/target, mob/user, delay)
 	if(!chassis)
 		return
 	var/C = chassis.loc
 	. = do_after(user, delay, target=target)
-	if(!chassis || chassis.loc != C || src != chassis.selected || !(get_dir(chassis, target)&chassis.dir))
+	if(!chassis || chassis.loc != C || !(get_dir(chassis, target)&chassis.dir))
 		return FALSE
 
-/obj/item/mecha_parts/mecha_equipment/proc/can_attach(obj/vehicle/sealed/mecha/M)
-	if(LAZYLEN(M.equipment)<M.max_equip)
+/obj/item/mecha_parts/mecha_equipment/proc/can_attach(obj/vehicle/sealed/mecha/M, rightclickattach = FALSE)
+	return default_can_attach(M, rightclickattach)
```

Comment:
snake_case for `rightclickattach` please. 
Also it might be clearer to rename to something like "attach_to_right" or something. `rightclickattach` isn't a very clear var name.

Reply:
Done